### DISCUSSION TITLE
bug: accessibility issues with popup modals

### DISCUSSION THREAD

**Discussion by SuhelMakkad** [supabase/supabase#16837] [Contribute to Supabase]
# Bug report

- [x] I confirm this is a bug with Supabase, not with my own application. 
- [x] I confirm I have searched the [Docs](https://docs.supabase.com), GitHub [Discussions](https://github.com/supabase/supabase/discussions), and [Discord](https://discord.supabase.com). 

## Describe the bug

Accessibility issues with popup modals. 
When I try navigating with the tab key, the focus is getting stuck on the model itself (see the below video)

## To Reproduce

Steps to reproduce the behavior, please provide code snippets or a repository:

1. Go to the authentication tab
2. Click on Users on the sidebar
3. Click on add user -> create new user
4. Hit the tab to go to the next input field
5. The focus will shift to the modal and then again come back to the input

## Expected behavior

When we hit the tab the focus should shift to the next input field or button.

## Screenshots

https://github.com/supabase/supabase/assets/71591136/e460ef65-8af2-4f50-b65a-6735ae2c7aa5

## System information

- OS: macOS
- Browser chrome

**Comment by Jonath-z** [comment]
@SuhelMakkad I would like to work on this issue, can you assign me?

**Comment by GaryAustin1** [comment]
@Jonath-z this was from one of the Supabase devs yesterday.
_Hey kishanhitk! We don't really have an assignment process. You can feel free to work on it though — adding an in-progress PR here often helps, even if it's very-very in-progress!_

**Comment by Jonath-z** [comment]
Great, thank you! I start working on it then

**Comment by ghost** [comment]
Hey @Jonath-z, I believe the issue is being caused by the **tabindex="-1"** attribute on modal **div**. If you remove this attribute, the problem should be resolved.

**Comment by Jonath-z** [comment]
I'm gonna do that and see.

**Comment by Jonath-z** [comment]
@nwaliaez do you have an idea about why I can not access the authentication page locally? 

This is the error that I'm getting:

![Screenshot from 2023-08-16 09-57-30](https://github.com/supabase/supabase/assets/79799120/04a2f548-9518-4bff-8a2c-717577c7da68)

**Comment by ghost** [comment]
@Jonath-z, based on my investigations, the issue seems to be related to PostgreSQL. To address this:

1. You'll need to set up PostgreSQL using the docker-compose.yml file located in the docker/ directory.
2. Before doing that, create a .env file within the docker/ directory.
3. Copy the contents of .env.example to the newly created .env file.

I'm currently trying this approach on my end. I'll update this thread once I have more information on whether it resolves the issue.

**Comment by Jonath-z** [comment]
@nwaliaez thanks.
 I investigated, I think is issue is caused by the Modal wrapper, as we can also see the same behavior on send invite model, How can I approach it?
 
 
[Screencast from 2023-08-16 14-16-34.webm](https://github.com/supabase/supabase/assets/79799120/ab0ccb12-57a0-4b31-807d-e76e724bf4f0)

# To reproduce

Steps to reproduce the behavior, please provide code snippets or a repository:

Go to the authentication tab
Click on Users on the sidebar
Click on add user -> send invitation
Hit the tab to go to the next input field
The focus will shift to the modal and then again come back to the input

**Comment by ghost** [comment]
@Jonath-z, I've made another observation. Once a valid string is entered into the input, the "TAB" key functions as expected.

**Comment by Jonath-z** [comment]
@nwaliaez 

hmm, I'm not able to get that, for me even with the righ input it still does not work as expected. Do you have an idea about the reason why?

**Comment by sagarkashyap989** [comment]
i am new here,  i am facing some issue while seting up=the project,
after cloning i tried `npm i` but its taking too long, is there is anything i am doing wrong.

**Comment by ghost** [comment]
@sagarkashyap989 Thats perfect it will take some time depending on network speed.

**Comment by sagarkashyap989** [comment]
i see, i wait then. (i actually came here after watching you video only).
thank you,
is it okay, if i ping you when i get stuck solving in the just this issue

**Comment by ghost** [comment]
@sagarkashyap989, I'm here to assist in any way I can. Feel free to reach out whenever you need help!

**Comment by sagarkashyap989** [comment]
After a clean git clone, I could not start & setup the project

Running` npm install` and i am facing these issuse:
![image](https://github.com/supabase/supabase/assets/75683778/c3b3fccb-d2bb-4e1a-8779-9e1146bf3bf3)

**Comment by ghost** [comment]
Hi @sagarkashyap989, you might want to consider these steps: clearing your cache, removing package-lock.json, and deleting node_modules. After that, giving 'yarn' a try worked like a charm for me. Feel free to give it a shot!

**Comment by justynleung** [comment]
I was testing this issue and I found out that if the user has a valid input in the input field, the tab button will navigate to the button. The problem was only reproduced without the valid input. I don't think it is necessary to fix this because an input is expected, it will then navigate to the button.

**Comment by ghost** [comment]
@justynleung Hello, I made another attempt, but even with valid input, the issue persists for me. Pressing the tab key still causes the modal to receive focus.

**Comment by justynleung** [comment]
@nwaliaez 
I think I have found the issue. Only invite user modal is not working.
For the add URL modal, it is actually working with a valid input. Could you please double check that too?

**Comment by ghost** [comment]
I've implemented a solution that appears to be effective when tested on my local machine. However, it's not yet functional on the website.

**Reply by ghost** [reply]
its working now XD

**Reply by justynleung** [reply]
I could not find a functional solution for the problem.
